### ISSUE TITLE
InvocationTX:  DecodeBinary bug (Master branch)

### ISSUE THREAD

**Issue by dauTT** [nspcc-dev/neo-go#173]
The implementaion of the `DecodeBinary` method is not correct:
https://github.com/CityOfZion/neo-go/blob/master/pkg/core/transaction/invocation.go#L36
```go
// DecodeBinary implements the Payload interface.
func (tx *InvocationTX) DecodeBinary(r io.Reader) error {
	lenScript := util.ReadVarUint(r)
	tx.Script = make([]byte, lenScript)
	if err := binary.Read(r, binary.LittleEndian, tx.Script); err != nil {
		return err
	}
	return binary.Read(r, binary.LittleEndian, &tx.Gas)
}
```

the equivalent C# looks as follow:
https://github.com/neo-project/neo/blob/master/neo/Network/P2P/Payloads/InvocationTransaction.cs#L24
```C#
protected override void DeserializeExclusiveData(BinaryReader reader)
        {
            if (Version > 1) throw new FormatException();
            Script = reader.ReadVarBytes(65536);
            if (Script.Length == 0) throw new FormatException();
            if (Version >= 1)
            {
                Gas = reader.ReadSerializable<Fixed8>();
                if (Gas < Fixed8.Zero) throw new FormatException();
            }
            else
            {
                Gas = Fixed8.Zero;
            }
        }
```

as we can see the golang implementation fails to include the `Version` logic.

For this raw transaction
```go
raw_tx:= "d1001b00046e616d6567d3d8602814a429a91afdbaa3914884a1c90c733101201cc9c05cefffe6cdd7b182816a9152ec218d2ec000000141403387ef7940a5764259621e655b3c621a6aafd869a611ad64adcc364d8dd1edf84e00a7f8b11b630a377eaef02791d1c289d711c08b7ad04ff0d6c9caca22cfe6232103cbb45da6072c14761c9da545749d9cfd863f860c351066d16df480602a2024c6ac"
byteTx, err := hex.DecodeString(param); err != nil {
			err = errors.Wrapf(err, "unable to decode %s to []byte", param.StringVal)
			resultsErr = NewInvalidParamsError(err.Error(), err)
		} else {
			r := bytes.NewReader(byteTx)
			tx := &transaction.Transaction{}
			err = tx.DecodeBinary(r)
}
```
we expect tx to be:
```go
{'txid': '0x1672df78b7dd21f3516fb0759518dfab29cbe106715504a59a3e12a359850397', 'size': 157, 'type': 'InvocationTransaction', 'version': 0, 'attributes': [{'usage': 32, 'data': '1cc9c05cefffe6cdd7b182816a9152ec218d2ec0'}], 'vout': [], 'vin': [], 'sys_fee': '0', 'net_fee': '0', 'scripts': [{'invocation': '403387ef7940a5764259621e655b3c621a6aafd869a611ad64adcc364d8dd1edf84e00a7f8b11b630a377eaef02791d1c289d711c08b7ad04ff0d6c9caca22cfe6', 'verification': '2103cbb45da6072c14761c9da545749d9cfd863f860c351066d16df480602a2024c6ac'}], 'script': '00046e616d6567d3d8602814a429a91afdbaa3914884a1c90c7331', 'gas': 0}
```

in particular `gas = 0`

**Comment by dauTT** [comment]
Closed issue due to inactivity.

**Comment by roman-khimov** [comment]
There is a testcase in `pkg/rpc/server_test.go` that still is failing (but it's commented out at the moment).

**Comment by roman-khimov** [comment]
And there is also one that I'm porting now from dev, maybe I'll fix it to make tests pass.

**Comment by roman-khimov** [comment]
Yeah, I have the fix (very similar to the one in #320) now in the queue.

**Comment by volekerb** [comment]
=) all right!
